siliconcompiler/tools/yosys/syn_fpga.tcl

```diff
@@ -4,19 +4,159 @@ set job_name [dict get $sc_cfg option jobname]
+#          real product part numbers and the if/else statement selecting a
+#          synthesis command.  It will be necessary to do some vetting and
+#          decide what to really support and what to omit based on the yosys
+#          support (or the place and route support) being too skimpy.
+#          -PG 1/13/2023
+
 if {[string match {ice*} $sc_partname]} {
     yosys synth_ice40 -top $sc_design -json "${sc_design}_netlist.json"
 } else {
-    yosys script "${build_dir}/${sc_design}/${job_name}/${step}/${index}/inputs/vpr_yosyslib/synthesis.ys"
+    #yosys script "${build_dir}/${sc_design}/${job_name}/${step}/${index}/inputs/vpr_yosyslib/synthesis.ys"
+    #Match VPR reference flow's hierarchy check, including their comments
+
+    #Here are the notes from the VPR developers
+    # These commands follow the generic `synth'
+    # command script inside Yosys
+    # The -libdir argument allows Yosys to search the current 
+    # directory for any definitions to modules it doesn't know
+    # about, such as hand-instantiated (not inferred) memories
+    #***NOTE:  Removed -libdir as it's nonsensical with the
+    #          use case of this script -PG 1/11/2023
+    yosys hierarchy -check -auto-top
+
+    #Rename top level module to match selected design name;
+    #This allows us to enforce some naming consistency in
+    #automated flows
+    #***NOTE:  because rename is a Tcl keyword the yosys -import
+    #          trick doesn't work
+    yosys rename -top ${sc_design}
+
+    yosys proc
+    #select -module ${sc_design}
+    #flatten -wb
+    #flatten
+
+    #Match the optimization passes of the VTR reference yosys flow
+    yosys opt_expr
+    yosys opt_clean
+    yosys check
+    yosys opt -nodffe -nosdff
+    yosys fsm
+    yosys opt
+    yosys wreduce
+    yosys peepopt
+    yosys opt_clean
+    yosys share
+    yosys opt
+    yosys memory -nomap
+    yosys opt -full
+
+    #Do our own thing from here
+    yosys techmap -map +/techmap.v
+
+    #***TO DO:  Figure out how to deal with this in an SC-compatible way
+    if { $sc_flop_techlib eq "None" } {
+	yosys dfflegalize -cell \$_DFF_P_ 01 -cell \$_DFF_P??_ 01 
+    }
+
+    #opt
+    #opt_clean
+
+    #Perform preliminary buffer insertion before passing to ABC to help reduce
+    #the overhead of final buffer insertion downstream
+    yosys insbuf
+
+    yosys abc -lut $sc_syn_lut_size -dff
+    #abc -script $env(FPGA_ARCHITECT_ROOT)/test/scripts/fpgaa_abc_flow.scr
+
+    #***TO DO:  Wrap this in a verbosity flag
+    #yosys write_verilog -noexpr -selected ${sc_design}_abc.v
+
+    #Clean up the design before buffer insertion to avoid blowing up the design
+    #size with that step
+    #opt
+    #clean
+
+    yosys flatten
+    yosys clean
+
+    #***NOTE:  If you flatten after this step instead of before Yosys barfs,
+    #          would be nice to have a deeper understanding of why that is
+    #          -PG 1/16/2023
```

Comment:
whats the issue on github for this in yosys repo?

Reply:
I have not filed a github issue for this at YosysHQ; nor am I aware of whether someone else has.